### ISSUE TITLE
pc.PRIMITIVE_POINTS does not work

### ISSUE THREAD

**Issue by willeastcott** [playcanvas/engine#1621]
PlayCanvas' GLSL never sets gl_PointSize. I'm assuming this means we can't correctly render points by setting a pc.Mesh's primtive type to pc.PRIMITIVE_POINTS.

We need public API to set point size. This would live on one of the following:

1. pc.Material
2. pc.Mesh (primitive object presumably)
3. pc.MeshInstance

Feels like 2 is the right choice but I could be convinced otherwise.

Origin of this issue is here:

https://github.com/playcanvas/playcanvas-gltf/issues/79

**Comment by aidinabedi** [comment]
If point size is added to the engine it should be done in the shader-code-generating materials (i.e. standard and basic), anything else can break app-specific custom shader code. And apps that already calculate their own point size in their shader may have their `gl_PointSize` overridden.

That being said, these shader-gen materials are primarily meant for actual surface areas, not lines or points. Devs should in IMHO, not be encouraged to use these advanced materials for rendering points, 
since there is little use for lighting, normals, textures etc. More than half of the functionality in the standard material will effectively waste GPU unless disabled when a point size is assigned.

In conclusion, I don't think there is much to gain in supporting point size on an engine-level. The glTF loader, however, should definitely support point size (defaulting to 1px) since that is in the specs.

**Comment by mvaligursky** [comment]
I agree it would not be useful to expose gl_PointSize in engine, as it's primarily useful in custom shaders. I've added a demo of custom shader and validated it's all functioning. Closing.
